cloud/cloudstack/cs_loadbalancer_rule_member.py

```diff
@@ -0,0 +1,253 @@
+    has_lib_cs = True
+except ImportError:
+    has_lib_cs = False
+
+# import cloudstack common
+from ansible.module_utils.cloudstack import *
+
+
+class AnsibleCloudStackLBRuleMember(AnsibleCloudStack):
+
+    def get_rule(self, **kwargs):
+        rules = self.cs.listLoadBalancerRules(**kwargs)
+        if rules:
+            return rules['loadbalancerrule'][0]
+
+    def _get_common_args(self):
+        return {
+            'account': self.get_account(key='name'),
+            'domainid': self.get_domain(key='id'),
+            'projectid': self.get_project(key='id'),
+            'zoneid': self.get_zone(key='id'),
+        }
+
+    def _change_members(self, operation):
+        if operation not in ['add', 'remove']:
+            raise RuntimeError('Bad operation: %s' % operation)
+        args = self._get_common_args()
+        rule = self.get_rule(name=self.module.params.get('name'), **args)
+        if not rule:
+            self.module.fail_json(msg="Unknown rule: %s" % args['name'])
+        if not self.module.check_mode:
+            res = self.cs.listLoadBalancerRuleInstances(id=rule['id'])
+            existing = {}
+            for vm in res.get('loadbalancerruleinstance', []):
+                existing[vm['name']] = vm['id']
+            wanted_names = self.module.params.get('vms')
+            if operation =='add':
+                cs_func = self.cs.assignToLoadBalancerRule
+                to_change = set(wanted_names) - set(existing.keys())
+            else:
+                cs_func = self.cs.removeFromLoadBalancerRule
+                to_change = set(wanted_names) & set(existing.keys())
+            if not to_change:
+                return rule
+            vms = self.cs.listVirtualMachines(**args)
+            to_change_ids = []
+            for name in to_change:
+                for vm in vms.get('virtualmachine', []):
+                    if vm['name'] == name:
+                        to_change_ids.append(vm['id'])
+                        break
+                else:
+                    self.module.fail_json(msg="Unknown VM: %s" % name)
+            res = cs_func(
+                id = rule['id'],
+                virtualmachineids = to_change_ids,
+            )
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+            poll_async = self.module.params.get('poll_async')
+            if poll_async:
+                self.poll_job(res)
+            self.result['changed'] = True
+        return rule
+
+    def add_members(self):
+        return self._change_members('add')
+
+    def remove_members(self):
+        return self._change_members('remove')
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(),
+            vms = dict(required=True),
+            state = dict(choices=['present', 'absent'], default='present'),
+            zone = dict(default=None),
```

Comment:
hmm but here https://github.com/ansible/ansible-modules-extras/pull/982/files#diff-3b35a6b36e38ac06c3856c928c352854R147 you wouldn't find the rule for a project right? IMHO project_id is neded also for _member